File: GFG/permutate.py
def permutate(s, l, r):
    if l == r:
        print(s)
    else:
        for i in range(1, r + 1):
            s[l], s[i] = s[i], s[l]
            permutate(s, l + 1, r)
            s[l], s[i] = s[i], s[l]

s = ['a', 'b', 'c']
permutate(s, 0, 2)

File: GFG/permutate.cpp
#include <bits/stdc++.h>

using namespace std;

void permutate(string s, int l, int r) {
    if (l == r) {
        cout << s << endl;
    }
    else {
        for (int i = l; i <= r; i++){
            swap(s[i], s[l]);
            permutate(s, l + 1, r);
            swap(s[i], s[l]);
        }
    }
}

int main() {
    string s = "abc";
    int n = s.size();
    permutate(s, 0, n - 1);
    return 0;
}